format/udf-spec.md

```diff
@@ -0,0 +1,402 @@
+| *optional*  | `secure`          | `boolean`              | Whether it is a secure function. Default: `false`.                                                              |
+| *optional*  | `doc`             | `string`               | Documentation string.                                                                                           |
+
+### Parameter-Name
+| Requirement | Field  | Type     | Description              |
+|-------------|--------|----------|--------------------------|
+| *required*  | `name` | `string` | Parameter name.          |
+| *optional*  | `doc`  | `string` | Parameter documentation. |
+
+Notes:
+1. When `secure` is `true`:
+   - Engines MUST NOT expose the function definition or its body through any form of metadata inspection (e.g., `SHOW FUNCTIONS`).
+   - Engines MUST prevent leakage of sensitive information during execution via error messages, logs, query plans, or intermediate results.
+   - Engines MUST NOT perform predicate reordering, short-circuiting, or other optimizations that could change the order or scope of data access.
+2. Entries in `properties` are treated as hints, not strict rules. Engines MAY choose to honor them or ignore them.
+3. `parameter-names` is the source of truth for parameter naming across all overload definitions, which makes named-argument
+   invocation consistent across definitions:
+   - Each overload uses the first N entries of this list for its arity, in order.
+   - Names and their relative ordering are immutable. Only appending new names is allowed.
+   - Only the `doc` field may be updated in place.
+   - Overloads that differ only by parameter types are allowed (e.g., `foo(int)`, `foo(float)`, `foo(string, string)`), and
+     they all derive their parameter names positionally from the same `parameter-names` prefix.
+
+### Definition
+
+Each `definition` represents one function signature (e.g., `add_one(int)` vs `add_one(float)`).
+
+| Requirement | Field name           | Type                                                                                                                                                                                                                                                        | Description                                                                                                                                                                                         |
+|-------------|----------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```

Comment:
Minor: In the table spec, we avoid table wrapping issues by omitting whitespace from the last field. That would help readability here.